include/dynd/types/string_type.hpp

```diff
@@ -22,6 +22,8 @@ class DYND_API string : public bytes {
 
   string(const std::string &other) : string(other.data(), other.size()) {}
 
+  string(const bytes &other) : string(other.data(), other.size()) {}
+
   bool operator<(const string &rhs) const
```

Comment:
Hey @mdboom thanks for getting into the code, it's awesome to see more faces in the commit logs :)

I think we want to keep a barrier between dynd::bytes and dynd::string, just like in Python 3. That would mean no implicit conversions, and structuring it so that the explicit conversions are very clear.